tests/test_ro_crate_parser.py

```diff
@@ -0,0 +1,137 @@
+# pylint: disable=missing-docstring
+# nosec assert_used
+from pathlib import Path
+
+import src.utils.filesystem.filters as filters
+from src.blueprints.datafile import RawDatafile
+from src.blueprints.dataset import RawDataset
+from src.blueprints.experiment import RawExperiment
+from src.blueprints.project import RawProject
+from src.extraction.manifest import IngestionManifest
+from src.profiles.ro_crate.ro_crate_parser import ROCrateParser
+from tests.fixtures.fixtures_ro_crate import (
+    fakecrate_root,
+    fixture_fake_ro_crate,
+    fixture_ro_crate_name,
+    fixture_rocrate_datafile,
+    fixture_rocrate_dataset,
+    fixture_rocrate_experiment,
+    fixture_rocrate_parser,
+    fixture_rocrate_project,
+    fixture_rocrate_uuid,
+    ro_crate_dataset_dir,
+    ro_crate_unlisted_file_dir,
+    rocrate_profile_json,
+    test_rocrate_content,
+)
+
+
+def test_load_crate(
+    fixture_rocrate_parser: ROCrateParser,
+    fakecrate_root: Path,
+    fixture_ro_crate_name: str,
+    fixture_rocrate_uuid: str,
+    ro_crate_unlisted_file_dir: Path,
+) -> None:
+    """Basic Test to load an entire RO-Crate from a fixture
+
+    Args:
+        fixture_rocrate_parser (ROCrateParser): a fixture of an already loaded parser
+            (based on other fixtures).
+        Test will fail if this parser fails to load.
+        fakecrate_root (Path): The location of the RO-Crate in the fake filesystem
+    """
+    assert fixture_rocrate_parser.name == fixture_ro_crate_name
+    assert fixture_rocrate_parser.uuid.hex == fixture_rocrate_uuid
+
+
+def test_read_crate_dataobjects(
+    fixture_rocrate_parser: ROCrateParser,
+    ro_crate_unlisted_file_dir: Path,
+    fixture_ro_crate_name: str,
+    fixture_rocrate_uuid: str,
+    fixture_rocrate_datafile: RawDatafile,
+    fixture_rocrate_dataset: RawDataset,
+) -> None:
+    """Test to confrim the RO-Crate parser has traversed datafiles correctly
+    and associated them with the correct datasets.
+    Also confirms that datasets haved been parsed correctly.
+    """
+    ingestible_dfs: IngestionManifest = fixture_rocrate_parser.process_datasets(
+        IngestionManifest(), filters.PathFilterSet(filter_system_files=True)
+    )
+    assert len([df.description for df in ingestible_dfs.get_datasets()]) == 2
```

Comment:
We're generating a temporary `list` of descriptions and checking its length, but won't its length be wholly determined by the length of the list from the manifest? So would it be equivalent just to write:
```python
assert len(ingestible_dfs.get_datasets()) == 2
```